cpp/src/fil/treelite_import.cu

```diff
@@ -430,8 +357,8 @@ int tree2fil_sparse(std::vector<fil_node_t>& nodes,
       // reserve space for child nodes
       // left is the offset of the left child node relative to the tree root
       // in the array of all nodes of the FIL sparse forest
-      int left = built_index - root;
-      built_index += 2;
+      int left = is_dense<fil_node_t>() ? 2 * cur + 1 : sparse_index - root;
+      sparse_index += 2;
       conversion_state<fil_node_t> cs = tl2fil_inner_node<fil_node_t>(
```

Comment:
I wonder if `tl2fil_inner_node()` and `tl2fil_leaf_payload()` would better be methods of `tl2fil_t`.

Feel free to do it in another pull request, though.